resizable-polyline/demo.js

```diff
@@ -0,0 +1,139 @@
+      '<circle cx="10" cy="10" r="7" fill="transparent" stroke="red" stroke-width="4"/>' +
+      '</svg>',
+      polyline = new H.map.Polyline(
+        new H.geo.LineString([51.6, 21.8, 0, 52.3, 24, 0, 51.7, 25.8, 0, 50.6, 26, 0, 50, 24.3, 0, 49.2, 23.2, 0, 48, 23.2, 0]),
+        {
+          style: {fillColor: 'rgba(150, 100, 0, .8)', lineWidth: 10}
+        }
+      ),
+      polylineGroup = new H.map.Group({
+        volatility: true, // mark the group as volatile for smooth dragging of all it's objects
+        objects: [polyline]
+      }),
+      polylineTimeout;
+
+  // ensure that the polyline can receive drag events
+  polyline.draggable = true;
+
+  // create markers for each polyline's node which will be used for dragging
+  polyline.getGeometry().eachLatLngAlt(function(lat, lng, alt, index) {
+    var marker = new H.map.Marker(
+      {lat, lng},
+      {
+        icon: new H.map.Icon(svgCircle, {anchor: {x: 10, y: 10}})
+      }
+    );
+    marker.setVisibility(false);
+    marker.draggable = true;
+    marker.setData({'nodeIndex': index})
+    polylineGroup.addObject(marker);
+  });
+
+  // add group with polyline and it's nodes (markers)
+  map.addObject(polylineGroup);
+
+  // event listener for polyline group to show markers if moved in with mouse (or touched on touch devices)
+  polylineGroup.addEventListener('pointerenter', function(evt) {
+    var group = this;
+    if (polylineTimeout) {
+      clearTimeout(polylineTimeout);
+      polylineTimeout = null;
+    }
+
+    // show markers
+    group.forEach(function(object) {
+      if (object instanceof H.map.Marker) {
+        object.setVisibility(true);
+      }
+    });
+  }, true);
+
+  // event listener for polyline group to hide markers if moved out with mouse (or released finger on touch devices)
+  // the outline is hidden on touch devices after specific timeout
+  polylineGroup.addEventListener('pointerleave', function(evt) {
+    var group = this,
+        timeout = (evt.currentPointer.type == 'touch') ? 1000 : 0;
+
+    polylineTimeout = setTimeout(function() {
+      group.forEach(function(object) {
+        if (object instanceof H.map.Marker) {
+          object.setVisibility(false);
+        }
+      });
+    }, timeout);
+
+    document.body.style.cursor = 'default';
+  }, true);
+
+  // event listener for polyline group to change the cursor if mouse position is over markers
+  polylineGroup.addEventListener('pointermove', function(evt) {
+    if (evt.target instanceof H.map.Marker) {
+      document.body.style.cursor = 'pointer';
+    } else {
+      document.body.style.cursor = 'default'
+    }
+  }, true);
+
+  // event listener for polyline group to resize the geo polyline object if dragging over markers
+  polylineGroup.addEventListener('drag', function(evt) {
+    var pointer = evt.currentPointer,
+        group = this,
+        polyline = group.getObjects()[0],
```

Comment:
The same like other example, I think it's better to use the outer scope.

Reply:
done